### DISCUSSION TITLE
Query().WithRelated() O2M as count?

### DISCUSSION THREAD

**Discussion by idc77** [ent/ent#2487] [Q&A]
I have an entity where I'd like to query the related 1:M entity but only get the count.
Let's get concrete:

```go
		posts := client.Post.Query().Where(post.BlogIDEQ(blog.ID)).
			WithAuthor().
			WithCategories().
			WithComments(). // those should return a count of comments of the post
			WithTags().
			AllX(cx)
```
I can of course leave this as is and do a `{{len(post.Edges.Comments)}}` but why pull all the data when all I need is the count in this case?

Is something like that possible with ent?

**Comment by idc77** [comment]
19 days no answer
I think if something is so complicated that using it makes your workflow more difficult and while you wait for no answer you're stuck, it's better to not use this piece of "technology" and stick with what's intuitive and what works. e.g. not ent.

ent seem to be more concerned about hype than functionality.

**Reply by lrstanley** [reply]
These types of comments are not beneficial to the conversation, [the community](https://github.com/ent/ent/blob/master/CODE_OF_CONDUCT.md), or the project. There are multiple other sources where you can receive support, in addition to Github Discussions -- if you don't get a response, there is more than one place you can go.

No ORM (or ORM-like) library/module/etc will ever provide exactly what you need, there are always gaps, or areas where the featureset doesn't match your requirements. This isn't exclusive of Ent, or Go, or of any similar thing written in other languages. You will have this same problem across the board. 

With that being said, what you're looking for is likely very easily doable. For example, take a look at [sql modifiers](https://entgo.io/docs/feature-flags#custom-sql-modifiers). There are likely other ways as well.
